nailgun/entities.py

```diff
@@ -9213,3 +9213,44 @@ def __init__(self, server_config=None, **kwargs):
             'read_type': 'base',
         }
         super().__init__(server_config=server_config, **kwargs)
+
+
+class AnsibleVariable(
+    Entity,
+    EntityCreateMixin,
+    EntityReadMixin,
+    EntityDeleteMixin,
+    EntitySearchMixin,
+    EntityUpdateMixin,
+):
+    """A representation of a Ansible Variable entity."""
+
+    def __init__(self, server_config=None, **kwargs):
+        self._fields = {
+            'variable': entity_fields.StringField(required=True),
+            'ansible_role_id': entity_fields.IntegerField(required=True),
+            'default_value': entity_fields.StringField(),
+            'override_value_order': entity_fields.StringField(),
```

Comment:
There is some diff between API and apidoc regarding the `hidden_value`. Apidoc says "ansible_variable[hidden_value]" but in response I can see we get `hidden_value?`.
```
2024-11-14 09:49:59 - nailgun.client - DEBUG - Received HTTP 200 response: {"parameter":"var_with_hidden_val","id":75,"variable":"var_with_hidden_val","ansible_role":"RedHatInsights.insights-client","ansible_role_id":1,"description":"","override":true,"variable_type":"string","hidden_value?":true,"validator_type":"","validator_rule":null,"merge_overrides":false,"merge_default":false,"avoid_duplicates":false,"override_value_order":"fqdn\nhostgroup\nos\ndomain","created_at":"2024-11-13 13:58:13 UTC","updated_at":"2024-11-13 13:58:13 UTC","default_value":"abc","imported":false,"override_values":[],"override_values_count":0}
```
So the change needed would be this, which looks quite weird
```suggestion
            'default_value': entity_fields.StringField(),
            'override_value_order': entity_fields.StringField(),
            'hidden_value?': entity_fields.BooleanField(),
```
@Gauravtalreja1 Can you check with DEVs what is expected and if they need to fix the API or apidoc?

Anyway, I wouldn't block the PR until it's resolved. It can be added anytime when needed.

Reply:
Seeing that the PR has been open for ~ 3 months I would not start to hurry now and clarify this in that PR.

Reply:
IMHO this is a bug https://github.com/theforeman/foreman_ansible/pull/744

Reply:
@Gauravtalreja1 @vsedmik As suggested by you above I would merge and cherry-pick the changes and then add the hidden value field to stream once the bug is fixed.

Reply:
@dosas Sorry for the late response. It seems your fix https://github.com/theforeman/foreman_ansible/pull/744 has been merged meanwhile. Do you want to update this PR with the `hidden_value` now?

Checked your fix locally:
```
$ curl -sku admin:changeme -H "Content-type: application/json" -X GET https://satellite.redhat.com/ansible/api/ansible_variables/11 | jq
{
  "parameter": "foreman_scap_client_apt_repo_url",
  "id": 11,
  "variable": "foreman_scap_client_apt_repo_url",
  "ansible_role": "theforeman.foreman_scap_client",
  "ansible_role_id": 2,
  "description": "",
  "override": true,
  "variable_type": "string",
  "validator_type": "",
  "validator_rule": null,
  "merge_overrides": false,
  "merge_default": false,
  "avoid_duplicates": false,
  "override_value_order": "fqdn\nhostgroup\nos\ndomain",
  "created_at": "2024-12-19 08:46:27 UTC",
  "updated_at": "2024-12-19 08:48:15 UTC",
  "default_value": "deb [trusted=yes] https://apt.atix.de/Ubuntu20LTS/ stable main",
  "imported": true,
  "override_values": [],
  "override_values_count": 0,
  "hidden_value": true   <---------------------
}
```